ts-client/src/dlmm/helpers/binArray.ts

```diff
@@ -322,3 +324,43 @@ export function findNextBinArrayWithLiquidity(
 
   return binArrayAccount;
 }
+
+/**
+ * Retrieves the bin arrays required to initialize multiple positions in continuous range.
+ *
+ * @param {PublicKey} pair - The public key of the pair.
+ * @param {BN} fromBinId - The starting bin ID.
+ * @param {BN} toBinId - The ending bin ID.
+ * @return {[{key: PublicKey, index: BN }]} An array of bin arrays required for the given position range.
+ */
+export function getBinArraysRequiredByPositionRange(
+  pair: PublicKey,
+  fromBinId: BN,
+  toBinId: BN,
+  programId: PublicKey
+): { key: PublicKey; index: BN }[] {
+  const [minBinId, maxBinId] = fromBinId.lt(toBinId)
+    ? [fromBinId, toBinId]
+    : [toBinId, fromBinId];
+
+  const positionCount = getPositionCount(minBinId, maxBinId);
```

Comment:
why we need position count here? 
We just need to get lowerBinArray index from fromBinID, and upperBinArray index from toBinId, and loop through this range to get all binArrays?

Reply:
If the bin range < 70 (`MAX_BIN_PER_POSITION`), it will the same bin arrays for lower and upper but when add / remove liquidity, the program require `lowerBinArrayIndex` and `lowerBinArrayIndex + 1`